onnxruntime/core/providers/nnapi/nnapi_builtin/builders/op_support_checker.cc

```diff
@@ -719,6 +719,41 @@ bool GemmOpSupportChecker::HasSupportedInputsImpl(const Node& node) const {
       });
 }
 
+// Get the bias size (C) of Gemm op
+// ANEURALNETWORKS_FULLY_CONNECTED only supports 1d bias
+// Will test if C of Gemm can be squeezed and return the 1d vector size after squeeze
+static bool GetBiasSize(const Shape& c_shape, int32_t android_sdk_ver, uint32_t& size) {
+  // TODO add support of scalar C for Gemm
+  if (c_shape.empty()) {
+    LOGS_DEFAULT(VERBOSE) << "C of Gemm cannot be a scalar";
+    return false;
+  }
+
+  if (c_shape.size() != 1 && android_sdk_ver < 28) {
+    LOGS_DEFAULT(VERBOSE) << "C of Gemm can only be 1d tensor for API level " << android_sdk_ver
+                          << " shape of C, " << Shape2String(c_shape);
+    return false;
+  }
+
+  size = c_shape[0];
+  bool size_assigned = false;
+  for (const auto& dim : c_shape) {
+    if (dim != 1) {
```

Comment:
Does this also need to check there are no dims with value 1 after size is assigned? You can't squeeze 1,3,1 to 3 can you as that would broadcast differently?

Reply:
Yes, we should only accept tensor in format {1,1,...,1,n} for C of Gemm